src/app/item-popup/SocketDetails.tsx

```diff
@@ -200,6 +205,14 @@ function SocketDetails({
         i.plug.energyCost.energyType === DestinyEnergyType.Any
     )
     .filter(isPluggableItem)
+    .filter(
+      (i) =>
```

Comment:
You could pull `isPluggableItem` into the next filter.

Reply:
Actually, I'm thinking about just removing the `isPluggableItem` filter because I think it's a no-op. At least in my profile response, `plug.canInsert` implies `plug.insertFailIndexes.length > 0`, for all plugs, so I don't think this is filtering out anything useful.

Reply:
Getting a type error in that case because `isPluggableItem` has a type guard that allows TS to refine the array. A more complex condition makes it opaque to the type checker.